src/databricks/labs/ucx/source_code/base.py

```diff
@@ -176,12 +176,22 @@ def lint(self, code: str) -> Iterable[Advice]:
     def lint_tree(self, tree: Tree) -> Iterable[Advice]: ...
 
 
-class Fixer:
+class Fixer(ABC):
     @abstractmethod
-    def name(self) -> str: ...
+    def can_fix(self, advice_code: str) -> bool: ...
+
+    @abstractmethod
+    def apply(self, advice_code: str, source_code: str) -> str: ...
```

Comment:
this is Fixer interface that has nothing to do with this PR. revert this change.

Reply:
It's in scope because SparkSqlPyLinter is both a linter and a fixer. It needs to route `apply` to the correct SQL fixer.